### PR TITLE
drivers: crypto: stm32: Replace buffer len assert by error logic

### PR DESCRIPTION
**Pull request by erwango**
In case asserts are deactivated, no check is done on buffers length. Remove asserts and return an error when lengths are not correct.

Check error in case length is set by API user.

### CODE REVIEW

**Review by gautierg-st — DISMISSED** [review]

File: drivers/crypto/crypto_stm32.c [hunk 1]
```diff
@@ -50,18 +50,22 @@ LOG_MODULE_REGISTER(crypto_stm32);
 
 struct crypto_stm32_session crypto_stm32_sessions[CRYPTO_MAX_SESSION];
 
-static void copy_reverse_words(uint8_t *dst_buf, int dst_len,
+static int copy_reverse_words(uint8_t *dst_buf, int dst_len,
 			       uint8_t *src_buf, int src_len)
```
Comment (mrfuchs) [inline drivers/crypto/crypto_stm32.c:54]:
```suggestion
			      uint8_t *src_buf, int src_len)
```

File: drivers/crypto/crypto_stm32.c [hunk 2]
```diff
@@ -176,6 +180,7 @@ static int crypto_stm32_cbc_encrypt(struct cipher_ctx *ctx,
 	struct crypto_stm32_session *session = CRYPTO_STM32_SESSN(ctx);
 
 	copy_reverse_words((uint8_t *)vec, sizeof(vec), iv, BLOCK_LEN_BYTES);
```
Comment (mrfuchs) [inline drivers/crypto/crypto_stm32.c:182]:
```suggestion
	(void)copy_reverse_words((uint8_t *)vec, sizeof(vec), iv, BLOCK_LEN_BYTES);
```

File: drivers/crypto/crypto_stm32.c [hunk 3]
```diff
@@ -203,6 +208,7 @@ static int crypto_stm32_cbc_decrypt(struct cipher_ctx *ctx,
...
```
Comment (mrfuchs) [inline drivers/crypto/crypto_stm32.c:210]:
```suggestion
	(void)copy_reverse_words((uint8_t *)vec, sizeof(vec), iv, BLOCK_LEN_BYTES);
```

**Review by mrfuchs — COMMENTED** [review]
Just some minor nits, otherwise LGTM.

File: drivers/crypto/crypto_stm32.c (hunk 1 above)
Comment (erwango) [reply drivers/crypto/crypto_stm32.c]: fixed, txs

**Review by mrfuchs — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**